homeassistant/components/velbus/quality_scale.yaml

```diff
@@ -0,0 +1,70 @@
+rules:
+  # Bronze
+  action-setup: todo
+  appropriate-polling:
+    status: exempt
+    comment: |
+      This integration does not poll.
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: todo
+  dependency-transparency: done
+  docs-actions: todo
+  docs-high-level-description: todo
+  docs-installation-instructions: todo
+  docs-removal-instructions: todo
+  entity-event-setup: todo
+  entity-unique-id: done
+  has-entity-name: todo
+  runtime-data: todo
+  test-before-configure: done
+  test-before-setup: todo
+  unique-config-entry: done
+
+  # Silver
+  action-exceptions: todo
+  config-entry-unloading: done
+  docs-configuration-parameters: todo
+  docs-installation-parameters: todo
+  entity-unavailable: todo
+  integration-owner: done
+  log-when-unavailable: done
+  parallel-updates: todo
+  reauthentication-flow:
+    status: exempt
+    comment: |
+      This integration does not require authentication.
+  test-coverage: todo
+  # Gold
+  devices: done
+  diagnostics: done
+  discovery-update-info: done
+  discovery: done
```

Comment:
So I understand that it currently works via USB, but in the `services.yaml` I also see examples with IP, so I am now assuming that you can connect to this bus either via USB or via local network. Did you check if the device:
- emits mDNS packets we can discover using zeroconf?
- Its mac address is a registered OUI?
- It has a default hostname it advertises with on the network?

Reply:
This all depends on the device that you connect to the bus.

*usb* modules => these are autodiscoverd via the discovery
*tcp/ip* => some modules have an mDNS (the newer ones that are not yet available on the market, they will be Q1 2025) others do not, so for some we can autodiscover, but not for all

Reply:
And what about the DHCP for the other ones? Like I do think its nice to autodiscover as much as possible :)

Reply:
how do you see this? not sure if dhcp would help here ...

Reply:
DHCP discovery works either by OUI or by hostname (or both). You can check the mac address of your device in https://www.wireshark.org/tools/oui-lookup.html for example. For example if I try to put my sonos speaker mac address in it, I find that sonos registered it. Depending on what the company makes, we can use that to assume a sonos device is at that IP and try to connect, and if that works, we can propose the user to add it. Same goes for hostname, sometimes devices join the network like `manufacturer-12ab34` as hostname, and then we can match on `manufacturer-*`

Reply:
hmm, all velbus its tcp/ip devices are basically rapberry pi modules
So i'm not sure if we can do something with it

Reply:
Ah if that is the case I don't think we can do that. Thanks for the info!